src/Raven.Server/Documents/Patch/ScriptRunner.cs

```diff
@@ -230,6 +237,174 @@ public SingleRun(DocumentDatabase database, RavenConfiguration configuration, Sc
+
+                string timeseries = GetStringArg(args[1], signature, "timeseries");
+                var timestamp = GetDateArg(args[2], signature, "timestamp");
+                var tag = GetStringArg(args[3], signature, "tag");
+                
+                if(args[4].IsArray() == false)
+                    throw new ArgumentException($"{signature}: The values should be an array but got {args[4].GetType().Name}");
+                var jsValues = args[4].AsArray();
+                var values = ArrayPool<double>.Shared.Rent((int)jsValues.Length);
+                try
+                {
+                    FillDoubleArrayFromJsArray(values, jsValues, signature);
+
+                    var toAppend = new TimeSeriesStorage.Reader.SingleResult
+                    {
+                        Values = new Memory<double>(values, 0, (int)jsValues.Length),
+                        Tag = _jsonCtx.GetLazyString(tag),
+                        Timestamp = timestamp,
+                        Status = TimeSeriesValuesSegment.Live
+                    };
+
+                    _database.DocumentsStorage.TimeSeriesStorage.AppendTimestamp(
+                        _docsCtx,
+                        id,
+                        CollectionName.GetCollectionName(doc),
+                        timeseries,
+                        new[] { toAppend});
+                }
+                finally                
+                {
+                    ArrayPool<double>.Shared.Return(values);
+                }
+                
+                return Undefined.Instance;
+            }
+
+            private JsValue DeleteRangeTimeSeries(JsValue self, JsValue[] args)
+            {
+                AssertValidDatabaseContext("deleteRangeTs");
+                
+                const string signature = "deleteRangeTs(doc, timeseries, from, to)";
+                const int requiredArgs = 4;
+                
+                if (args.Length != requiredArgs)
+                    throw new ArgumentException($"{signature}: This method requires {requiredArgs} arguments but was called with {args.Length}");
+                
+                var (id, doc) = GetIdAndDocFromArg(args[0], signature);
+
+                string timeseries = GetStringArg(args[1], signature, "timeseries");
+
+                var from = GetDateArg(args[2], signature, "from");
+                var to = GetDateArg(args[3], signature, "to");
+
+                var deletionRangeRequest = new TimeSeriesStorage.DeletionRangeRequest
+                {
+                    DocumentId = id,
+                    Collection = CollectionName.GetCollectionName(doc),
+                    Name = timeseries,
+                    From = from,
+                    To = to,
+                };
+                _database.DocumentsStorage.TimeSeriesStorage.RemoveTimestampRange(_docsCtx, deletionRangeRequest);
+                
+                return JsValue.Undefined;
```

Comment:
Why are we returning this? What should be returned here? Do we need to return anything?